exercises/grep/tests/grep.rs

```diff
@@ -0,0 +1,483 @@
+extern crate grep;
+
+use grep::grep;
+
+use std::{
+    fs, sync::{Once, ONCE_INIT},
+};
+
+static ILIAD_CONTENT: &'static str = "Achilles sing, O Goddess! Peleus' son;
+His wrath pernicious, who ten thousand woes
+Caused to Achaia's host, sent many a soul
+Illustrious into Ades premature,
+And Heroes gave (so stood the will of Jove)
+To dogs and to all ravening fowls a prey,
+When fierce dispute had separated once
+The noble Chief Achilles from the son
+Of Atreus, Agamemnon, King of men.
+";
+
+static MIDSUMMER_NIGHT_CONTENT: &'static str = "I do entreat your grace to pardon me.
+I know not by what power I am made bold,
+Nor how it may concern my modesty,
+In such a presence here to plead my thoughts;
+But I beseech your grace that I may know
+The worst that may befall me in this case,
+If I refuse to wed Demetrius.
+";
+
+static PARADISE_LOST_CONTENT: &'static str = "Of Mans First Disobedience, and the Fruit
+Of that Forbidden Tree, whose mortal tast
+Brought Death into the World, and all our woe,
+With loss of Eden, till one greater Man
+Restore us, and regain the blissful Seat,
+Sing Heav'nly Muse, that on the secret top
+Of Oreb, or of Sinai, didst inspire
+That Shepherd, who first taught the chosen Seed
+";
+
+static BLOK_CONTENT: &'static str = "Белой ночью месяц красный
+Выплывает в синеве.
+Бродит призрачно-прекрасный,
+Отражается в Неве.
+Мне провидится и снится
+Исполпенье тайных дум.
+В вас ли доброе таится,
+Красный месяц, тихий шум?..
+";
+
+struct Fixture;
+
+impl Fixture {
+    fn set_up(&self) {
+        static SETUP: Once = ONCE_INIT;
+
+        SETUP.call_once(|| {
+            set_up_files(&vec![
+                "iliad.txt",
+                "midsummer_night.txt",
+                "paradise_lost.txt",
+                "blok.txt",
+            ]);
+        });
+    }
+}
+
+impl Drop for Fixture {
+    fn drop(&mut self) {
+        /*tear_down_files(&vec![
```

Comment:
This is one of the main problems: if I remove the comment now because of `const` copy behavior the tests will enter the data race and once one test will run, it will delete the files and other tests would not be able to access them. I really want to make the test fixture test-independent, with files creating at the start of test run and deleting them after all tests are complete

Reply:
I see what you mean, and I agree: making the fixtures test-independent really is more elegant than leaving files hanging around the filesystem. However, that's tough. I've fiddled around with it a bit, and agree that actually getting the tests to delete the file fixtures without running into race conditions is hard.

One option would be to use [galvanic-test](https://github.com/mindsbackyard/galvanic-test) or something similar to handle the fixtures for us. However, that goes against the exercism ethos that the exercises should have minimal external dependencies. 

I think the most straightforward solution to this problem in the end will be to just remove the `Drop` implementation, and leave the files hanging around on the user's filesystem. It's not great, but it's what we can do within the constraints we're given.

Reply:
Another solution is to append test names to file names, so every test creates and works with it's own set of files. It is not a great approach in terms of adding new tests but it can work for this exercise

Reply:
Yes, that might work.